components/script/dom/htmlinputelement.rs

```diff
@@ -134,6 +142,26 @@ impl HTMLInputElement {
             .get_attribute(&ns!(), &atom!("type"))
             .map_or_else(|| atom!(""), |a| a.value().as_atom().to_owned())
     }
+
+    fn get_value_mode(&self) -> ValueMode {
+        HTMLInputElement::get_value_mode_for_input_type(self.input_type.get())
+    }
+
+    // https://html.spec.whatwg.org/multipage/#input-type-attr-summary
+    fn get_value_mode_for_input_type(input_type: InputType) -> ValueMode {
```

Comment:
Why is this written in a functional style? The `HTMLInputElement` has all the necessary information about its own input type.